samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -82,17 +82,19 @@
    * non-stream systems (e.g., an external database).
    *
    * @param sinkFn the function to send messages in this stream to an external system
+   * @return this {@link MessageStream} object
    */
-  void sink(SinkFunction<M> sinkFn);
+  MessageStream<M> sink(SinkFunction<? super M> sinkFn);
```

Comment:
Sold on this one. Will keep sink / sendTo terminal. The implementation in PR is more like a side output. Maybe we can add later as sideSink()/sideSend(), if more and more use cases popping up.